modules/services/repositories/src/main/java/au/com/shiftyjelly/pocketcasts/repositories/podcast/SuggestedFoldersManager.kt

```diff
@@ -0,0 +1,43 @@
+package au.com.shiftyjelly.pocketcasts.repositories.podcast
+
+import au.com.shiftyjelly.pocketcasts.models.db.AppDatabase
+import au.com.shiftyjelly.pocketcasts.models.db.dao.SuggestedFoldersDao
+import au.com.shiftyjelly.pocketcasts.models.entity.SuggestedFolder
+import au.com.shiftyjelly.pocketcasts.servers.podcast.PodcastCacheServiceManager
+import au.com.shiftyjelly.pocketcasts.servers.podcast.SuggestedFoldersRequest
+import jakarta.inject.Inject
+import kotlin.coroutines.CoroutineContext
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.flow.Flow
+import timber.log.Timber
+
+class SuggestedFoldersManager @Inject constructor(
+    private val podcastCacheService: PodcastCacheServiceManager,
+    appDatabase: AppDatabase,
+) : CoroutineScope {
+
+    private val suggestedFoldersDao: SuggestedFoldersDao = appDatabase.suggestedFoldersDao()
+
+    override val coroutineContext: CoroutineContext
+        get() = Dispatchers.Default
+
+    fun getSuggestedFolders(): Flow<List<SuggestedFolder>>? {
+        try {
+            return suggestedFoldersDao.findAll()
+        } catch (e: Exception) {
+            Timber.e(e, "Getting suggested folders failed")
+            return null
+        }
+    }
+
+    suspend fun refreshSuggestedFolders(podcastUuids: List<String>) {
+        try {
+            val request = SuggestedFoldersRequest(podcastUuids)
+            val folders: List<SuggestedFolder> = podcastCacheService.suggestedFolders(request)
+            suggestedFoldersDao.insertAll(folders)
```

Comment:
This implementation results in these scenarios:
* When a different set of folders is returned from the server the old suggestions persist.
* When a different set of podcasts is returned for the safe folder the old podcasts are still present.

This can lead to us suggesting wrong folders or folders with wrong podcasts.

Reply:
humm... Good point. So I think we should delete the existing one and add the new folders if the server returns something

Reply:
@MiSikora I've pushed the changes [d5b56f3](https://github.com/Automattic/pocket-casts-android/pull/3592/commits/d5b56f36e64cc4320e7f83fd1528b8295822ee81)